opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -467,6 +481,48 @@ static int mca_btl_tcp_component_close(void)
     return OPAL_SUCCESS;
 }
 
+/*
+ * Set an appropriate bandwidth for the btl instance. On Linux, we obtain this
+ * via an ioctl(). All other OSs, we return MCA_BTL_TCP_BTL_BANDWIDTH.
+ */
+static unsigned int
+mca_btl_tcp_ethtool_get_speed (const char *if_name)
+{
+#if !HAVE_DECL_SIOCETHTOOL
+    return MCA_BTL_TCP_BTL_BANDWIDTH;
+#else
+    int sockfd;
+    struct ifreq ifr;
+    struct ethtool_cmd edata = {
+        .cmd = ETHTOOL_GSET,
+    };
+    unsigned int speed = 0;
+
+    sockfd = socket(PF_INET, SOCK_DGRAM, 0);
+    if (sockfd < 0) {
+        goto out;
+    }
+
+    memset(&ifr, 0, sizeof(struct ifreq));
+    strncpy(ifr.ifr_name, if_name, IF_NAMESIZE); /* TODO: IFNAMSIZ instead? */
+    ifr.ifr_data = (char *)&edata;
+
+    if (ioctl(sockfd, SIOCETHTOOL, &ifr) < 0) {
+        goto out;
+    }
+
+    speed = ethtool_cmd_speed(&edata);
```

Comment:
Do we need to check for the presence of this `ethtool_cmd_speed` function?  E.g., see http://stackoverflow.com/questions/26031152/linux-ethtool-command